mathesar/database/utils.py

```diff
@@ -21,3 +22,20 @@ def get_database_key(engine):
 
 def get_non_default_database_keys():
     return [key for key in settings.DATABASES if key != 'default']
+
+
+def update_databases():
+    databases = set(settings.DATABASES)
+
+    # Update deleted databases
+    for database in Database.objects.all():
+        if database.name in databases:
+            databases.remove(database.name)
+        else:
+            database.deleted = True
+            Schema.objects.filter(database=database).delete()
```

Comment:
I assume this also deletes `Table` objects.

Reply:
Yep, confirmed in the database api tests.